### DISCUSSION TITLE
How to save selected video background effect in "CallWithChatComposite"

### DISCUSSION THREAD

**Discussion by lovedota** [Azure/communication-ui-library#5450] [Q&A]
Hi

I am using CallWithChatComposite
```
const callAdapterOptions: AzureCommunicationCallAdapterOptions = useMemo(() => {
    return {
      videoBackgroundOptions: {
        videoBackgroundImages: VIDEO_BACKGROUND_IMAGES,
        onResolveDependency: onResolveVideoEffectDependencyLazy
      },
    };
  }, []);
```
Do we have any callback when users change the selected video effect, then we can save the key in LocalStorage and set it for users everytime they join the meeting by default.

**Comment by mgamis-msft** [comment]
Hi @lovedota,

I checked our code base and it looks we currently do not have a callback/listener on the adapter that triggers when a video effect is selected. I will confirm with my team and whether or not we can add callback/listener for this.

**Comment by mgamis-msft** [comment]
@lovedota While the selected background effect has no specific callback/listener, you can use onStateChange to track it like shown in this [show and tell discussion](https://github.com/Azure/communication-ui-library/discussions/5453) I created.

Would this work for you even though it might be a little hacky? Or would a better API for listening to selected background effect be more helpful?
